File: easy/n-ary-tree-preorder-traversal/n-ary-tree-preorder-traversal.cpp
class Solution {
public:
    vector<int> res;
    void dfs(Node* root) {
        if (root) {
            res.push_back(root->val);
            for (auto& i : root->children) {
                dfs(i);
            }
        }
    }
    vector<int> preorder(Node* root) {
        dfs(root);
        return res;
    }
};

File: easy/n-ary-tree-preorder-traversal/n-ary-tree-preorder-traversal.py
class Solution:
    def preorder(self, root: 'Node') -> List[int]:
        res=[]
        def dfs(node):
            if node:
                res.append(node.val)
                for i in node.children:
                    dfs(i)
        dfs(root)
        return res
